client/src/pages/ContractorSignUp.js

```diff
@@ -0,0 +1,148 @@
+import React, { useState } from "react";
+
+function ContractorSignUp() {
+	const [contractorDetails, setContractorDetails] = useState({
+		userType: "contractor",
+		firstName: "",
+		lastName: "",
+		email: "",
+		company: "",
+		address: "",
+	});
+	const [emailError, setEmailError] = useState("");
+	// This variable shows success of the request to during sign up that comes back from server
+	const [resgisterStatus, setRegisterStatus] = useState("");
+
+	const validateEmail = (email) => {
+		const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+		return emailRegex.test(email);
+	};
+
+	async function postContractorDetails(userData) {
+		try {
+			const response = await fetch("/mock-api", {
+				method: "POST",
+				headers: {
+					"Content-Type": "application/json",
+				},
+				body: JSON.stringify(userData),
+			});
+			const data = response.json();
+			if (!response.ok) {
+				setRegisterStatus(data.message);
+			}
+			setRegisterStatus(data.message);
+		} catch (error) {
+			setRegisterStatus(error.message);
```

Comment:
Don't show the raw error message to the user. It might contain technical details that they don't understand. Just tell them there was an error.